handler/src/main/java/io/netty/handler/ssl/util/FingerprintTrustManagerFactory.java

```diff
@@ -135,45 +131,97 @@ private void checkTrusted(String type, X509Certificate[] chain) throws Certifica
 
     /**
      * Creates a new instance.
+     * <p>
+     * <strong>NOTE:</strong> This deprecated constructor uses SHA-1 that is considered insecure.
+     * It is recommended to specify a stronger hash algorithm, such as SHA-256,
+     * by calling {@link FingerprintTrustManagerFactory#builder(String)} method.
+     * </p>
      *
      * @param fingerprints a list of SHA1 fingerprints in hexadecimal form
      */
+    @Deprecated
     public FingerprintTrustManagerFactory(Iterable<String> fingerprints) {
-        this(toFingerprintArray(fingerprints));
+        this("SHA1", toFingerprintArray(fingerprints));
     }
 
     /**
      * Creates a new instance.
+     * <p>
+     * <strong>NOTE:</strong> This deprecated constructor uses SHA-1 that is considered insecure.
+     * It is recommended to specify a stronger hash algorithm, such as SHA-256,
+     * by calling {@link FingerprintTrustManagerFactory#builder(String)} method.
+     * </p>
      *
      * @param fingerprints a list of SHA1 fingerprints in hexadecimal form
      */
+    @Deprecated
     public FingerprintTrustManagerFactory(String... fingerprints) {
-        this(toFingerprintArray(Arrays.asList(fingerprints)));
+        this("SHA1", toFingerprintArray(Arrays.asList(fingerprints)));
     }
 
     /**
      * Creates a new instance.
+     * <p>
+     * <strong>NOTE:</strong> This deprecated constructor uses SHA-1 that is considered insecure.
+     * It is recommended to specify a stronger hash algorithm, such as SHA-256,
+     * by calling {@link FingerprintTrustManagerFactory#builder(String)} method.
+     * </p>
      *
      * @param fingerprints a list of SHA1 fingerprints
      */
+    @Deprecated
     public FingerprintTrustManagerFactory(byte[]... fingerprints) {
+        this("SHA1", fingerprints);
+    }
+
+    /**
+     * Creates a new instance.
+     *
+     * @param algorithm a hash algorithm
+     * @param fingerprints a list of fingerprints
+     */
+    private FingerprintTrustManagerFactory(final String algorithm, byte[][] fingerprints) {
+        ObjectUtil.checkNotNull(algorithm, "algorithm");
         ObjectUtil.checkNotNull(fingerprints, "fingerprints");
 
+        this.tlmd = new FastThreadLocal<MessageDigest>() {
+
+            @Override
+            protected MessageDigest initialValue() {
+                try {
+                    return MessageDigest.getInstance(algorithm);
+                } catch (NoSuchAlgorithmException e) {
+                    throw new IllegalArgumentException("Unsupported hash algorithm", e);
+                }
+            }
+        };
+
+        int hashLength = tlmd.get().getDigestLength();
```

Comment:
see above... I think it is very unlikely that the "constructing" thread will use the factory itself. so I think it would be better to not waste resources in the `FastThreadLocal`  and construct it directly.

Reply:
If I understand you correctly, you're suggesting getting rid of `FastThreadLocal`. I was not sure about it since I am new to the code. Okay, I can remove `FastThreadLocal` and create a `MessageDigest` directly in the constructor if it's fine.

Reply:
no ... What I was saying that you should keep the `FastThreadLocal` but not access it in the constructor but better just create a new instance in the constructor and use it here.

Reply:
Okay, sure I can create a `MessageDigest` in the constructor as an early-check that the algorithm exists. The initialization of `FastThreadLocal` would still stay in the constructor since it now requires an algorithm identifier.